src/kete_core/src/io/parquet.rs

```diff
@@ -0,0 +1,146 @@
+//! Optional feature support for reading/writing parquet table
+//!
+
+use itertools::Itertools;
+use std::fs::File;
+
+use crate::errors::{Error, KeteResult};
+use crate::frames::Frame;
+use crate::state::State;
+
+use polars::prelude::*;
+
+/// Write a collection of states to a parquet table.
+pub fn write_states_parquet(states: &[State], filename: &str) -> KeteResult<()> {
+    let desigs = Series::new(
```

Comment:
Unfortunately, because parquet is a columns based format, here we basically clone the entire states array into column vectors. This is inefficient, but it is an infrequent cost and easy to implement.